flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/runner/JobDispatcherLeaderProcess.java

```diff
@@ -34,14 +37,21 @@
 
     private final JobGraph jobGraph;
 
+    private final Collection<JobResult> recoveredDirtyJobResults;
+    private final JobResultStore jobResultStore;
+
     JobDispatcherLeaderProcess(
             UUID leaderSessionId,
             DispatcherGatewayServiceFactory dispatcherGatewayServiceFactory,
             JobGraph jobGraph,
+            Collection<JobResult> recoveredDirtyJobResults,
```

Comment:
Would not having a collection here make sense? We already do the same thing with the jobGraph (recoveredJobs collection in other implementations)
```suggestion
            @Nullable JobResult recoveredDirtyJobResult,
```